src/legacy/ui/public/saved_objects/components/saved_object_finder.tsx

```diff
@@ -52,31 +51,38 @@ interface SavedObjectFinderUIState {
   sortDirection?: Direction;
 }
 
-interface SavedObjectFinderUIProps extends InjectedIntlProps {
+interface BaseSavedObjectFinder {
   callToActionButton?: React.ReactNode;
   onChoose?: (
     id: SavedObject<SavedObjectAttributes>['id'],
     type: SavedObject<SavedObjectAttributes>['type']
   ) => void;
   makeUrl?: (id: SavedObject<SavedObjectAttributes>['id']) => void;
   noItemsMessage?: React.ReactNode;
-  savedObjectType: 'visualization' | 'search';
+  savedObjectType: 'visualization' | 'search' | 'index-pattern';
   visTypes?: VisTypesRegistryProvider;
+}
+
+interface SavedObjectFinderFixedPage extends BaseSavedObjectFinder {
+  initialPageSize?: undefined;
+  fixedPageSize: number;
```

Comment:
This seems more complicated than necessary - why not adding another prop `hidePerPageOptions` which defaults to `false`?

Reply:
We had that solution beforehand. It actually doesn't ease the code here at all, since we still must make sure, once that flag is set, that `initialPageSize` can be any value, but only can be `5 | 10 | 15` in case that flag is not set. So it wouldn't really change a thing here, and I found the wording that way a bit more explicit, that you know you're specifying the actual page size and not just an initial one.

Reply:
Makes sense, thanks. TBH I think we should just get rid of the page size dropdown for this component altogether.

Reply:
I think once we're switching over to using `EuiListGroup` we can anyway reevaluate what we need in it :)